### PR TITLE
chore: Adopt progress reporter API

### PR DESCRIPTION
**Pull request by jdneo**

### CODE REVIEW

File: src/commands/runFromUri.ts [hunk 1]
```diff
@@ -25,13 +27,19 @@ export async function executeTestsFromUri(uri: Uri | undefined, isDebug: boolean
         return;
     }
 
+    if (!progressReporter) {
+        progressReporter = progressProvider?.createProgressReporter(isDebug ? 'Debug Test' : 'Run Test', ProgressLocation.Notification, true);
```
Comment (testforstephen) [inline src/commands/runFromUri.ts:31]: progressReporter = progressReporter || progressProvider?.createProgressReporter(isDebug ? 'Debug Test' : 'Run Test', ProgressLocation.Notification, true);

File: src/extension.ts [hunk 2]
```diff
@@ -76,7 +77,15 @@ async function doActivate(_operationId: string, context: ExtensionContext): Prom
...
-        javaLanguageSupportVersion = extension.packageJSON.version;
+        javaLanguageSupportVersion = javaLanguageSupport.packageJSON.version;
+    }
+
+    const javaDebugger: Extension<any> | undefined = extensions.getExtension('vscjava.vscode-java-debug');
+    if (javaDebugger && javaDebugger.isActive) {
+        const extensionApi: any = javaDebugger.exports;
+        if (extensionApi) {
+            progressProvider = extensionApi.progressProvider;
+        }
```
Comment (testforstephen) [inline src/extension.ts:88]: progressProvider = javaDebugger.exports?.progressProvider;

File: src/runners/runnerScheduler.ts [hunk 3]
```diff
@@ -39,35 +41,43 @@ class RunnerScheduler {
...
         if (this._isRunning) {
             window.showInformationMessage('A test session is currently running. Please wait until it finishes.\n');
             return;
         }
 
         this._isRunning = true;
+
+        if (!progressReporter) {
+            progressReporter = progressProvider?.createProgressReporter(runnerContext.isDebug ? 'Debug Test' : 'Run Test', ProgressLocation.Notification, true);
+        }
```
Comment (testforstephen) [inline src/runners/runnerScheduler.ts:54]: progressReporter = progressReporter || progressProvider?.createProgressReporter(runnerContext.isDebug ? 'Debug Test' : 'Run Test', ProgressLocation.Notification, true);

File: src/extension.ts [hunk 4]
```diff
@@ -43,10 +44,10 @@ async function doActivate(_operationId: string, context: ExtensionContext): Prom
     await fse.ensureDir(storagePath);
     logger.initialize(storagePath, context.subscriptions);
 
-    const extension: Extension<any> | undefined = extensions.getExtension('redhat.java');
+    const javaLanguageSupport: Extension<any> | undefined = extensions.getExtension('redhat.java');
     let javaLanguageSupportVersion: string = '0.0.0';
-    if (extension && extension.isActive) {
-        const extensionApi: any = extension.exports;
+    if (javaLanguageSupport && javaLanguageSupport.isActive) {
```
Comment (Eskibear) [inline src/extension.ts:49]: `javaLanguageSupport?.isActive` ?

**Review by Eskibear — APPROVED** [review]

File: src/extension.ts (hunk 4 above)
Comment (jdneo) [reply src/extension.ts:49]: ok

**Review by Eskibear — APPROVED** [review]

**Review by testforstephen — APPROVED** [review]

**Pull request merged**